pinot-common/src/main/java/com/linkedin/pinot/common/http/MultiGetRequest.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (C) 2014-2015 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.common.http;
+
+import com.google.common.base.Preconditions;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletionService;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorCompletionService;
+import javax.annotation.Nonnull;
+import org.apache.commons.httpclient.HttpClient;
+import org.apache.commons.httpclient.HttpConnectionManager;
+import org.apache.commons.httpclient.methods.GetMethod;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Class to support multiple http GET operations in parallel by using
+ * the executor that is passed in.
+ *
+ * This is a wrapper around Apache common HTTP client.
+ *
+ * The execute method is re-usable but there is no real benefit to it. All
+ * the connection management is handled by the input HttpConnectionManager.
+ * For access through multiple-threads, use MultiThreadedHttpConnectionManager
+ * as shown in the example below.
+ *
+ * Usage:
+ * <pre>
+ * {@code
+ *    List<String> urls = Arrays.asList("http://www.linkedin.com", "http://www.google.com");
+ *    MultiGetRequest mget = new MultiGetRequest(Executors.newCachedThreadPool(),
+ *           new MultiThreadedHttpConnectionManager());
+ *    CompletionService<GetMethod> completionService = mget.execute(urls);
+ *    for (int i = 0; i < urls.size(); i++) {
```

Comment:
Is it possible to know the response corresponding to a URL? Do we care for our use case?